localstack/aws/connect.py

```diff
@@ -319,6 +320,10 @@ def _get_client(
         :param config: Boto config for advanced use.
         :return: Boto3 client.
         """
+        # Disable proxies for LocalStack-internal calls because boto3 does not support NO_PROXY anymore and
+        # proxying internal calls never makes sense for customers.
+        config = config.merge(botocore.client.Config(proxies={}))
+
```

Comment:
It makes sense here to avoid cache misses but this is now also enabled for external + clients against aws.
We might need to find another solution if someone needs proxying for those, at some point but this is fine for now!

Reply:
Mhmm, that basically breaks shared Lambda layer fetching in a proxy setup 😒 


What if we just patch the `no_proxy` config in `localstack/config.py` for now?

```
no_proxy = ",".join([LOCALHOST_HOSTNAME, LOCALHOST, LOCALHOST_IP, "[::1]"])
```

`LOCALSTACK_HOSTNAME` resolves to `localhost.localstack.cloud` and that seemed to be the missing part in this issue.

I still don't fully understand why `NO_PROXY` works because I thought it is not supported anymore but my tests showed that actually works 🤔 😕

Reply:
I searched through the site-packages and these are the libs that read `NO_PROXY`:

```
cd localstack/.venv/lib/python3.10/site-packages
ag NO_PROXY
boto/connection.py
697:        self.no_proxy = os.environ.get('no_proxy', '') or os.environ.get('NO_PROXY', '')

websocket/_url.py
75:DEFAULT_NO_PROXY_HOST = ["localhost", "127.0.0.1"]
106:        v = os.environ.get("no_proxy", os.environ.get("NO_PROXY", "")).replace(" ", "")
110:        no_proxy = DEFAULT_NO_PROXY_HOST

docker/utils/proxy.py
53:            env['no_proxy'] = env['NO_PROXY'] = self.no_proxy

requests/sessions.py
306:        NO_PROXY, we strip the proxy configuration. Otherwise, we set missing

requests/utils.py
860:    such a NO_PROXY to strip proxy configurations.

pip/_vendor/requests/sessions.py
306:        NO_PROXY, we strip the proxy configuration. Otherwise, we set missing

pip/_vendor/requests/utils.py
860:    such a NO_PROXY to strip proxy configurations.

localstack/config.py
508:elif os.environ.get("NO_PROXY"):
509:    os.environ["NO_PROXY"] += "," + no_proxy

future/backports/urllib/request.py
2423:    no_proxy = os.environ.get('no_proxy', '') or os.environ.get('NO_PROXY', '')
```

Reply:
I think it's in `botocore.utils`: https://github.com/boto/botocore/blob/63d5d3503f1b2267a10433e42727395729b039d0/botocore/utils.py#L2952